roles/pulp_webserver/templates/pulp-vhost.conf.j2

```diff
@@ -78,8 +78,8 @@ Define pulp-api {{ pulp_api_bind }}
 
   # most pulp static files are served through whitenoise
   # http://whitenoise.evans.io/en/stable/
-  ProxyPass / http://${pulp-api}/
-  ProxyPassReverse / http://${pulp-api}/
+  #ProxyPass / http://${pulp-api}/
+  #ProxyPassReverse / http://${pulp-api}/
```

Comment:
pulp2 request were falling here `/pulp/api/v2`

Reply:
it would fix, but how we would address the static files?

Reply:
@bmbouter @dkliban this is the second issue, it depends on relying on whitenoise, not so easy to fix

Reply:
The recent testing showed that if a more specific rule were present that it would take precedence over this rule. So I don't think this line being here is the actual root cause of the issue.

Reply:
Current config:
Request for pulp2: `/pulp/api/v2` would fall on (1)

1)
```nginx
  ProxyPass /pulp/api http://${pulp-api}/pulp/api
  ProxyPassReverse /pulp/api http://${pulp-api}/pulp/api
```
2)
```nginx
  ProxyPass /auth/login/ http://${pulp-api}/auth/login/
  ProxyPassReverse /auth/login/ http://${pulp-api}/auth/login/
```
3)
```ngnix
  ProxyPass /pulp/content http://${pulp-content}/pulp/content
  ProxyPassReverse /pulp/content http://${pulp-content}/pulp/content
```
4)
```nginx
  ProxyPass / http://${pulp-api}/
  ProxyPassReverse / http://${pulp-api}/
```

Addressing (1), Request for pulp2: `/pulp/api/v2` would fall on (4)

1)
```nginx
  ProxyPass /pulp/api/v3 http://${pulp-api}/pulp/api/v3
  ProxyPassReverse /pulp/api/v3 http://${pulp-api}/pulp/api/v3
```
2)
```nginx
  ProxyPass /auth/login/ http://${pulp-api}/auth/login/
  ProxyPassReverse /auth/login/ http://${pulp-api}/auth/login/
```
3)
```ngnix
  ProxyPass /pulp/content http://${pulp-content}/pulp/content
  ProxyPassReverse /pulp/content http://${pulp-content}/pulp/content
```
4)
```nginx
  ProxyPass / http://${pulp-api}/
  ProxyPassReverse / http://${pulp-api}/
```

Reply:
I don't understand what this comment is saying. Can you explain it some more to me? My interest is focusing on the current problem with the current state (assume no changes).

Reply:
I'm just pointing which rule would match with `/pulp/api/v2`, in our current config it is (1), fixing (1) it would match rule (4)

Reply:
I'm still not understanding. If we have a `/pulp/api/v3` rule in pulp3 and a `/pulp/api/v2/` rule in pulp2, these routes don't seem conflicting. If we have a `/pulp/api/v3` in pulp3 and a `/pulp/api/` in pulp2 the "specific routes first" matching causes these also to not be conflicting. Similarly, `/pulp/api/v2/` in pulp2 and `/pulp/api/` in pulp3. Basically I'm not understanding the root cause of the issue or even what the issue is in practice.

Reply:
actually, the pulp2 config is totally different than the pulp3 one, I'm not sure if we have a `/pulp/api/` or `/pulp/api/v2` rule there. I posted pulp2 config below here: https://github.com/pulp/pulp_installer/pull/402#issuecomment-677835818
The only mention `pulp/api` mention I see is:
```apache
WSGIScriptAlias /pulp/api /usr/share/pulp/wsgi/webservices.wsgi
```
And honestly, I don't know the behavior of `WSGIScriptAlias`

Reply:
> actually, the pulp2 config is totally different than the pulp3 one, I'm not sure if we have a `/pulp/api/` or `/pulp/api/v2` rule there. I posted pulp2 config below here: [#402 (comment)](https://github.com/pulp/pulp_installer/pull/402#issuecomment-677835818)
> The only `pulp/api` mention I see is:
> 
> ```apache-config
> WSGIScriptAlias /pulp/api /usr/share/pulp/wsgi/webservices.wsgi
> ```
> 
> And honestly, I don't know the behavior of `WSGIScriptAlias`

That's ok!

The info you've shared is helpful. Can you describe what the experience the user has that is an issue? What error or incorrect behaviour do they experience? Also how can that be reproduced?

Reply:
it is here: https://github.com/pulp/pulp_installer/pull/402#issuecomment-677833836
pulp2 requests go to pulp3